fix(common): Warn on fill ngOptimizedImage without height

atcastle opened the pull request:
This PR includes one final fix for the ngOptimizedImage directive. Currently it does not warn if the image is in fill mode and the rendered height of the image is 0. This is a fairly common failure case when starting out with the image directive, which is caused by not properly setting the `position` style of the parent element. 

The warning added in this PR will help new users diagnose why their fill image is not appearing on the page.

**Note:** I am currently unable to run (and therefor unable to author) e2e tests on my machine, and this feature can only be tested in an e2e context. @AndrewKushnir, @pkozlowski-opensource  or @kara, could one of you help me get an e2e test added for this change? It would follow the basic pattern used in `packages/core/test/bundling/image-directive/e2e/image-distortion/image-distortion.e2e-spec.ts`

atcastle committed:
fix(common): Warn on fill ngOptimizedImage without height

Add a warning if a fill-mode image is rendered without height. This is a common occurence if the user doesn't properly set the 'position' attribute of the parent element.

kara commented on packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts:
```diff
@@ -862,6 +863,26 @@ function assertEmptyWidthAndHeight(dir: NgOptimizedImage) {
+    dir: NgOptimizedImage, img: HTMLImageElement, renderer: Renderer2) {
+  const removeListenerFn = renderer.listen(img, 'load', () => {
+    removeListenerFn();
+    const renderedHeight = parseFloat(img.clientHeight as any);
```
Nit: can we replace `any` with a more specific type? e.g. `string|null`

pkozlowski-opensource replied:
more importantly: why do we need `parseFloat `? Seems like the type of `HTMLImageElement:clientHeight` already: https://www.typescriptlang.org/play?#code/MYewdgzgLgBAlgWwOYwLwzAVwDbZgQwgLAE8CiAJAFQFkAZASQXyQFMBRbVhVsKAbgBQg0JFgALNPGQA6YNji8oFVnCTiBwgA74AThFYAxbCHxQAFOICU-IA

atcastle replied:
Yeah, I think the parseFloat pattern got pasted around in a few places where it's not needed. I just added a cleanup task to the tracker to go back through and review typing and usages of parseFloat().

AndrewKushnir replied:
Thanks for the review Pawel! Since Alex added a cleanup item to the tracking doc, I'm moving this PR to the merge queue, so we include the fix into the v15 release.

kara reviewed (approved):
LGTM

AndrewKushnir commented:
[Presubmit](http://test/OCL:487906249:BASE:487906292:1668202927171:f762eae6).

AndrewKushnir committed:
test(common): add e2e tests for the fill mode checks in the NgOptimizedImage directive

This commit adds extra e2e tests for the fill mode checks in the NgOptimizedImage directive to make
sure a warning is logged in a console.

pkozlowski-opensource commented:
LGTM but left a cleanup comment.

atcastle commented:
LGTM! Thanks for your help with the e2e tests

jessicajaniuk commented:
This PR was merged into the repository by commit d6dc6b6349f3e0d3d92068ac00dfcac5579f0953.
